hooks/tk-multi-publish2/ingest/collector.py

```diff
@@ -0,0 +1,257 @@
+            properties["work_path_template"] = work_path_template
+
+        # build the context of the item
+        context = self._get_item_context_from_path(parent_item, properties, os.path.basename(path))
+
+        # create and populate the item
+        file_item = parent_item.create_item(
+            item_type,
+            type_display,
+            display_name,
+            collector=self.plugin,
+            context=context,
+            properties=properties
+        )
+
+        # resolve work_path_template for the item
+        file_item.properties["work_path_template"] = self._resolve_work_path_template(file_item.properties,
+                                                                                      os.path.basename(path))
+
+        # Set the icon path
+        file_item.set_icon_from_path(icon_path)
+
+        # if the supplied path is an image, use the path as the thumbnail.
+        if (item_type.startswith("file.image") or
+            item_type.startswith("file.texture") or
+            item_type.startswith("file.render")):
+
+            if is_sequence:
+                file_item.set_thumbnail_from_path(seq_files[0])
+            else:
+                file_item.set_thumbnail_from_path(path)
+
+            # disable thumbnail creation since we get it for free
+            file_item.thumbnail_enabled = False
+
+        if is_sequence:
+            # include an indicator that this is an image sequence and the known
+            # file that belongs to this sequence
+            file_info = (
+                "The following files were collected:<br>"
+                "<pre>%s</pre>" % (pprint.pformat(seq_files),)
+            )
+        else:
+            file_info = (
+                "The following file was collected:<br>"
+                "<pre>%s</pre>" % (path,)
+            )
+
+        self.logger.info(
+            "Collected item: %s" % display_name,
+            extra={
+                "action_show_more_info": {
+                    "label": "Show File(s)",
+                    "tooltip": "Show the collected file(s)",
+                    "text": file_info
+                }
+            }
+        )
+
+        return file_item
+
+    def on_context_changed(self, settings, item):
+        """
+        Callback to update the item on context changes.
+
+        :param dict settings: Configured settings for this collector
+        :param item: The Item instance
+        """
+
+        path = item.properties["path"]
+        is_sequence = item.properties["is_sequence"]
+
+        item_info = self._get_item_info(settings, path, is_sequence)
+
+        item.properties["work_path_template"] = self._resolve_work_path_template(item_info,
+                                                                                 os.path.basename(path))
+
+        # Set the item's fields property
+        item.properties["fields"] = self._resolve_item_fields(item)
+
+
+    def _resolve_item_fields(self, item):
+        """
```

Comment:
Why is this method overridden?

Reply:
I did you one better and broke out the template parsing part into a new overwrite-able sub-method `_get_template_fields_from_path()`, so you can just do something similar to what you did for `_resolve_work_path_template()`. Its available in `sgtk_apps-2.7.2`